src/data/metadata/mappers/PackageMapper.ts

```diff
@@ -0,0 +1,78 @@
+import _ from "lodash";
+
+export interface PackageTransformationStrategy<Input, Output> {
+    apiVersion: number;
+    transform(payload: Input): Output;
+}
+
+/**
+ * Apply consecutive transformations to domain package until bigger version transformation found for dhsi2 version
+ * if exists transformations for versions 30,31,33 and version argument is 31 then
+ * transformations 30 and 31 will be applied
+ * @param version version until apply transformations
+ * @param payload  payload to trasnform
+ * @param transformations list of possible transformations to apply
+ */
+export function mapPackageToD2<Input, Output>(
+    d2Version: number,
+    payload: Input,
+    transformations: PackageTransformationStrategy<unknown, Output>[] = []
+): Output {
+    const orderedTransformations = _.sortBy(transformations, "version");
+
+    const transformationstoApply = orderedTransformations.filter(
+        transformation => transformation.apiVersion <= d2Version
+    );
+
+    if (transformationstoApply.length > 0) {
+        return applyTransformations<Input, Output>(payload, transformations);
+    } else {
+        console.log(
+            `No transformations applied from domain to dhis2 metadata package in version ${d2Version}`
+        );
+        return (payload as unknown) as Output;
+    }
+}
+
+/**
+ * Apply consecutive transformations to dhis2 package until lower version transformation found for dhsi2 version that
+ * transform to domain metadata package
+ * if exists transformations for versions 30,31,33 and version argument is 31 then
+ * transformations 30 and 31 will be applied
+ * @param version version until apply transformations
+ * @param payload  payload to trasnform
+ * @param transformations list of possible transformations to apply
+ */
+export function mapD2PackageFromD2<Input, Output>(
+    d2Version: number,
+    payload: Input,
+    transformations: PackageTransformationStrategy<unknown, Output>[] = []
+): Output {
+    const orderedTransformations = _.sortBy(transformations, "version", "desc");
+
+    const transformationstoApply = orderedTransformations.filter(
+        transformation => transformation.apiVersion <= d2Version
+    );
+
+    if (transformationstoApply.length > 0) {
+        return applyTransformations<Input, Output>(payload, transformations);
+    } else {
+        console.log(
+            `No transformations applied from dhis2 to domain metadata package in version ${d2Version}`
+        );
+        return (payload as unknown) as Output;
+    }
+}
```

Comment:
This code was not working as expected, and even though the tests were passing it did not apply the correct transformations. I've fixed this in the next PR but leaving some comments on what I changed:

- _.sortBy second parameter is usually an array of properties that should be used for sorting (you can sort by multiple properties when exact results are found)
- _.sortBy does not accept ``desc`` as order, instead _.orderBy should be used
- Seems that ``transformationstoApply`` was not being used since ``applyTransformations`` is called with the parameter ``transformations``

As I said, there's nothing to change in this PR as everything has changed in the next one.